modules/tinymce/src/core/test/ts/browser/UndoManagerTest.ts

```diff
@@ -611,4 +612,80 @@ describe('browser.tinymce.core.UndoManagerTest', () => {
 
     assert.equal(changeEventCounter, 0, 'No events should be detected');
   });
+
+  context('dispatchChange', () => {
+    const initialContent = '<p>some inital content</p>';
+    const manualModifiedLevel = 'a modified last level';
+
+    const assertChangeEvent = (
+      event: { level: string; lastLevel: string },
+      expectedLevel: string | undefined,
+      expectedLastlevel: string | undefined
+    ) => {
+      assert.equal(event.level, expectedLevel, 'Level has not the expected content');
+      assert.equal(event.lastLevel, expectedLastlevel, 'Last level has not the expected content');
```

Comment:
These really should be using `deepEqual` as we don't need them to be the same reference, only the same data.

Reply:
I should add if we don't use the level then can we please change the variable names as we aren't passing the level here at the moment, only the content.

Reply:
done, I also changed the type of the function since the change event could pass undefined as a `lastLevel`

Reply:
@lorenzo-pomili I can't see these changes, as it's still got the misleading names and isn't passing the full level only the content. Did you forget to push them?

Reply:
Sorry, I was sure that I also changed the names of expectation parameters and the name of the keys of the input object. (fixed with the last commits)

But I'm probably still missing the point about the use of `deepEqual` + use directly the `UndoLevel`, in this test the expectations are strings so making `deepEqual` with `UndoLevel` would mean to recreate an `UndoLevel` from the expectation string.

If we want to pass the entire event to the function it could be refactored in this way:
```TypeScript
const assertChangeEvent = (
      event: { level: UndoLevel; lastLevel: UndoLevel | undefined } | undefined,
      expectedLevelContent: string | undefined,
      expectedLastlevelContent: string | undefined
    ) => {
      assert.equal(event?.level?.content, expectedLevelContent, 'Level has not the expected content');
      assert.equal(event?.lastLevel?.content, expectedLastlevelContent, 'Last level has not the expected content');
    };

    let changeEventCounter: number;
    let currentChangeEvent: { level: UndoLevel; lastLevel: UndoLevel | undefined } | undefined;

    const onChange = (e: EditorEvent<{
      level: UndoLevel;
      lastLevel: UndoLevel | undefined;
    }>) => {
      changeEventCounter++;
      currentChangeEvent = e;
    };
```

Reply:
Yes, that's the point I was referring to. As for the `deepEqual` part, you're right I've got that a bit mixed up sorry. I instead was thinking of doing a partial deep equal which isn't possible out of the box. Anyways, what you've got now is fine minus a little typo. If you preferred, you really could just go back to using `assert.equal` for each level content sorry.

Reply:
refactored, in the way that I wrote in the previous message